### PR TITLE
Move logic for adding trains to geometry::helper

### PR DESCRIPTION
**Pull request by schroedtert**
Instead of having the logic for computing the projected door coordinates onto the track walls and the splitting of the track walls in the TrainEvent the functionality has been moved to the geometry helper namespace. 

- ComputeTrainDoorCoordinates: function for computing the projected door coordinates onto the trackwalls
- AddTrainDoors: Interface for the train event to update the subroom, delete or add some walls and add the train doors
- SplitWalls: Split the walls on the given cutting points, returns walls to add/remove

Additionally the container for saving temporarily deleted/added walls, doors for trains have been made private for the Building, and corresponding getter/setter have been added.

WIP: 
- [x] improved unittests for the SplitWalls method   
- [x] missing documentation

Solves #742

### CODE REVIEW

File: libcore/src/events/TrainEvent.cpp [hunk 1]
```diff
@@ -240,80 +89,89 @@ void TrainEvent::TrainDeparture()
...
-        auto wall = *it;
-        if(it != tempWalls.end()) {
-            tempWalls.erase(it);
+    auto tempWalls = _building->GetTrainAddedWalls(_trainID);
+    if(tempWalls.has_value()) {
+        for(auto it = tempWalls.value().begin(); it != tempWalls.value().end();) {
+            auto wall = *it;
+            if(it != tempWalls.value().end()) {
+                tempWalls.value().erase(it);
+            }
```
Comment (zeroset) [inline libcore/src/events/TrainEvent.cpp:98]: This seems to be quite dangerous, as i am not sure if the iterator is still valid afterwards. Will this be changed in following PRs?

File: libcore/src/geometry/Building.cpp [hunk 2]
```diff
@@ -1252,4 +1187,100 @@ std::vector<TrainType> Building::GetTrainTypes()
...
+        _trainAddedWalls.emplace(trainID, std::vector<Wall>{trainAddedWall});
+    }
+}
+
+void Building::SetTrainAddedWalls(int trainID, std::vector<Wall> trainAddedWalls)
+{
+    _trainAddedWalls[trainID] = trainAddedWalls;
+}
+
+std::optional<std::vector<Wall>> Building::GetTrainAddedWalls(int trainID)
```
Comment (zeroset) [inline libcore/src/geometry/Building.cpp:1206]: Can we name this `GetTrainWallsAdded` to sound more like the walls added from the train? Or do i misunderstand?

File: libcore/src/events/TrainEvent.cpp (hunk 1 above)
Comment (schroedtert) [reply libcore/src/events/TrainEvent.cpp:100]: It is, will be tackled soon when solving #745

File: libcore/src/geometry/Building.cpp (hunk 2 above)
Comment (schroedtert) [reply libcore/src/geometry/Building.cpp]: Changed, also changed the names of the other functions to fit the naming

**Pull request merged**